src/crimson/common/exception.h

```diff
@@ -18,8 +18,21 @@ class system_shutdown_exception final : public std::exception{
   }
 };
 
+class crushmap_changed final : public std::exception{
+public:
+  crushmap_changed(bool sp) : still_primary(sp) {}
+  const char* what() const noexcept final {
+    return "acting set changed";
+  }
+  bool is_primary() {
+    return still_primary;
+  }
+private:
+  bool still_primary = false;
```

Comment:
could mark this member variable `const`. as it never changes after being set in the ctor.